lib/internal/net.js

```diff
@@ -5,7 +5,9 @@ module.exports = { isLegalPort };
 // Check that the port number is not NaN when coerced to a number,
 // is an integer and that it falls within the legal range of port numbers.
 function isLegalPort(port) {
-  if (typeof port === 'string' && port.trim() === '')
+  if (port === null) return true;
```

Comment:
wasn't sure if null should be rejected or treated as equivalent to 0. I'm happy rejecting nulls also.